### PR TITLE
#376 :bug: - Support interpolation for external profile properties

### PR DESCRIPTION
**Pull request by CedricMtta**

### CODE REVIEW

**Comment by slawekjaranowski** [comment]
You can try to add your reproduce example project as IT

File: src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java [hunk 1]
```diff
@@ -803,7 +810,13 @@ private ModelBuildingRequest createModelBuildingRequest(File pomFile) {
                 project.getRemoteProjectRepositories(),
                 getReactorModelsFromSession());
         Properties userProperties = this.session.getUserProperties();
+        settings.getProfiles()
+                .stream()
+                .filter(p -> settings.getActiveProfiles().contains(p.getId()))
+                .forEach(activeProfile -> userProperties.putAll(activeProfile.getProperties()));
+
         List<String> activeProfiles = this.session.getRequest().getActiveProfiles();
+        activeProfiles.addAll(this.settings.getActiveProfiles());
```
Comment (slawekjaranowski) [inline src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java:819]: I'm not sure if we try to modify list returned from request.
In Maven 4 most of objects will be immutable.

Comment (CedricMtta) [reply src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java]: I created local var instead of modifying the ones from the Maven objects

**Comment by CedricMtta** [comment]
> You can try to add your reproduce example project as IT

It's done

File: src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java [hunk 2]
```diff
@@ -802,11 +809,20 @@ private ModelBuildingRequest createModelBuildingRequest(File pomFile) {
...
                 project.getRemoteProjectRepositories(),
                 getReactorModelsFromSession());
-        Properties userProperties = this.session.getUserProperties();
-        List<String> activeProfiles = this.session.getRequest().getActiveProfiles();
+        Properties userAndActiveExternalProfilesProperties = new Properties();
+        userAndActiveExternalProfilesProperties.putAll(this.session.getUserProperties());
+        settings.getProfiles().stream()
+                .filter(p -> settings.getActiveProfiles().contains(p.getId()))
+                .forEach(
+                        activeProfile -> userAndActiveExternalProfilesProperties.putAll(activeProfile.getProperties()));
```
Comment (slawekjaranowski) [inline src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java:817]: We also have `session.getRequest().getInactiveProfiles()` 
https://maven.apache.org/guides/introduction/introduction-to-profiles.html#deactivating-a-profile

so profile id should be on ActiveProfiles from session request or from settings and not in InactiveProfiles list when build properties list

File: src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java [hunk 3]
```diff
@@ -802,11 +809,20 @@ private ModelBuildingRequest createModelBuildingRequest(File pomFile) {
...
+
+        List<String> activeProfiles = Stream.concat(
+                        this.session.getRequest().getActiveProfiles().stream(),
+                        this.settings.getActiveProfiles().stream())
+                .collect(Collectors.toList());
 
         return new DefaultModelBuildingRequest()
-                .setUserProperties(userProperties)
+                .setUserProperties(userAndActiveExternalProfilesProperties)
```
Comment (slawekjaranowski) [inline src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java:825]: There is also method `DefaultModelBuildingRequest.setInactiveProfileIds`, we shoud add

```
new DefaultModelBuildingRequest()
....
  .setInactiveProfileIds(session.getRequest().getInactiveProfiles())
```

File: src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java (hunk 2 above)
Comment (CedricMtta) [reply src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java]: Good catch, done 👍

File: src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java (hunk 3 above)
Comment (CedricMtta) [reply src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java:826]: Done 👍

**Pull request merged**